### PR TITLE
Add management of empty array error in json validator

### PR DESCRIPTION
**Pull request by pthegner**
# Description

This PR change the error message to override the default validator message "[] is too short"

## Issue

Issue link: bot-749

## How to test

Here are the following steps to test this pull request:

- POST / PUT disruption and POST / PUT impact with "impacts":[]
- POST / PUT disruption and POST / PUT impact with "objects":[]
- POST / PUT disruption and POST / PUT impact with "application_periods":[]
... (you can use also all array that have this validation : minItems = 1 -> see chaos\formats.py)

### CODE REVIEW

File: tests/features/create-disruption-with-impacts.feature [hunk 1]
```diff
@@ -150,6 +150,17 @@ Feature: Create Disruption and impacts
...
+        I fill in header "X-Coverage" with "jdr"
+        I fill in header "Authorization" with "d5b0148c-36f4-443c-9818-1f2f74a00be0"
+        When I post to "/disruptions" with:
+        """
+        {"reference": "foo", "contributor": "contrib1", "cause":{"id": "7ffab230-3d48-4eea-aa2c-22f8680230b6"}, "publication_period":{"begin":"2018-09-11T13:50:00Z","end":"2018-12-31T16:50:00Z"}, "impacts": [{"severity": {"id": "7ffab232-3d48-4eea-aa2c-22f8680230b6"}, "application_periods": [{"begin": "2014-04-29T16:52:00Z","end": "2014-06-22T02:15:00Z"},{"begin": "2014-04-29T16:52:00Z","end": "2014-05-22T02:15:00Z"}], "objects": [{"id":"line:JDR:M5", "type":"line_section","line_section": {"line":{"id":"line:JDR:M5","type":"line"}, "start_point":{"id":"stop_area:JDR:SA:BASTI", "type":"stop_area"}, "end_point":{"id":"stop_area:JDR:SA:CHVIN", "type":"stop_area"}, "sens":2, "routes":[{"type":"route", "id":"route:JDR:M14"}, {"type":"route", "id":"route:JDR:M1"}], "metas":[] }}]}]}
+        """
+        Then the status code should be "400"
+        And the header "Content-Type" should be "application/json"
+        And the field "error.message" should be "metas should not be empty"
```
Comment (dvdn) [inline tests/features/create-disruption-with-impacts.feature]: _metas_ isn't a required item

File: tests/features/impacts-with-pt-objects.feature [hunk 2]
```diff
@@ -1090,4 +1090,96 @@ Feature: Manipulate impacts in a Disruption
...
+        I fill in header "X-Coverage" with "jdr"
+        I fill in header "Authorization" with "d5b0148c-36f4-443c-9818-1f2f74a00be0"
+        When I put to "/disruptions/7ffab230-3d48-4eea-aa2c-22f8680230b6/impacts/7ffab232-3d47-4eea-aa2c-22f8680230b6" with:
+        """
+        {"severity": {"id": "7ffab232-3d48-4eea-aa2c-22f8680230b6"}, "objects": [{"id":"line:JDR:M5", "type":"line_section","line_section": {"line":{"id":"line:JDR:M5","type":"line"}, "start_point":{"id":"stop_area:JDR:SA:BASTI", "type":"stop_area"}, "end_point":{"id":"stop_area:JDR:SA:CHVIN", "type":"stop_area"}, "sens":2, "routes":[{"type":"route", "id":"route:JDR:M14"}, {"type":"route", "id":"route:JDR:M1"}], "metas":[] }}], "application_periods": [{"begin": "2014-04-29T16:52:00Z","end": "2014-06-22T02:15:00Z"}]}
+        """
+
+        Then the status code should be "400"
+        And the header "Content-Type" should be "application/json"
+        And the field "error.message" should be "metas should not be empty"
```
Comment (dvdn) [inline tests/features/impacts-with-pt-objects.feature]: https://github.com/CanalTP/Chaos/pull/392/files#diff-101ee5cf6a226535f91de9e8147af695R162 (already tested ?)

File: tests/features/update-cause.feature [hunk 3]
```diff
@@ -88,6 +88,14 @@ Feature: update cause
         Then the status code should be "400"
         And the header "Content-Type" should be "application/json"
 
+        When I put to "/causes/7ffab230-3d48-4eea-aa2c-22f8680230b6" with:
+        """
+        {"wordings":[]}
+        """
+        Then the status code should be "400"
+        And the header "Content-Type" should be "application/json"
+        And the field "error.message" should be "wordings should not be empty"
```
Comment (dvdn) [inline tests/features/update-cause.feature]: _wordings_ isn't  required item, but as its interesting. perhaps update documentation (swagger)

File: tests/features/update-severity.feature [hunk 4]
```diff
@@ -92,6 +92,15 @@ Feature: update severity
...
         And the header "Content-Type" should be "application/json"
 
+        I fill in header "X-Customer-Id" with "5"
+        When I put to "/severities/7ffab230-3d48-4eea-aa2c-22f8680230b6" with:
+        """
+        {"wordings": [], "color": "foo"}
+        """
+        Then the status code should be "400"
+        And the header "Content-Type" should be "application/json"
+        And the field "error.message" should be "wordings should not be empty"
```
Comment (dvdn) [inline tests/features/update-severity.feature]: Same as https://github.com/CanalTP/Chaos/pull/392/files#diff-c2a26be584b8513534e8706a7ab9cad8R97

File: tests/features/create-disruption-with-impacts.feature (hunk 1 above)
Comment (pthegner) [reply tests/features/create-disruption-with-impacts.feature]: But in the formatted code, it have a `minItems: 1`  (https://github.com/CanalTP/Chaos/blob/master/chaos/formats.py#L127) when it's a line_section.
What we test here is the response when this occurs : `minItems: 1`  not true, which response is "[] is too short".

File: tests/features/update-cause.feature (hunk 3 above)
Comment (pthegner) [reply tests/features/update-cause.feature]: wordings is in fact a required item (see post on severity and cause).

File: tests/features/update-severity.feature (hunk 4 above)
Comment (pthegner) [reply tests/features/update-severity.feature]: same response as before

File: tests/features/impacts-with-pt-objects.feature (hunk 2 above)
Comment (pthegner) [reply tests/features/impacts-with-pt-objects.feature]: We could delete it if you want. It's just that the test before enter another route that this test.

Comment (Fahmus) [reply tests/features/impacts-with-pt-objects.feature]: yes we prefer only one tests :)

**Pull request merged**